### PR TITLE
VPC subnet resource implmentation

### PR DESCRIPTION
**Pull request by ksamoray**
PR contains two commits - the first has the autogenerated files and the second, all the manual edits required to make it work.
As the generator is still being debugged, I'd rather keep it so - so I can regenerate whenever something is fixed in the generator (at least of as long as this PR isn't merged).

### CODE REVIEW

**Comment by annakhm** [comment]
LGTM but I think we need more test coverage - with `advanced_config`, `dhcp_config` and alternating presence of those

**Comment by salv-orlando** [comment]
@ksamoray can you update the target branch for this PR to vpc-feature-branch

File: nsxt/resource_nsxt_vpc_subnet.go [hunk 1]
```diff
@@ -209,13 +213,13 @@ func resourceNsxtVpcSubnet() *schema.Resource {
 		Update: resourceNsxtVpcSubnetUpdate,
 		Delete: resourceNsxtVpcSubnetDelete,
 		Importer: &schema.ResourceImporter{
-			State: nsxtParentPathResourceImporter,
+			State: nsxtVPCPathResourceImporter,
```
Comment (salv-orlando) [inline nsxt/resource_nsxt_vpc_subnet.go:216]: If this has already been fixed in the generator maybe we can try and update the generated commit so we don't have to manually change the importer?

Comment (ksamoray) [reply nsxt/resource_nsxt_vpc_subnet.go:216]: If I understand correctly, `nsxtParentPathResourceImporter` is useful for objects which have sub-objects of a level VPC obj, e.g `VpcSubnetPort` which is nested under `VpcSubnet` and therefore has  a parent object path.
IDK if we can conclude automatically from the spec if this is the case or not. @annakhm do you know?

File: nsxt/resource_nsxt_vpc_subnet.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ var vpcSubnetSchema = map[string]*metadata.ExtendedSchema{
 	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
 	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
 	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
-	"context":      metadata.GetExtendedSchema(getContextSchema(false, false, true)),
+	"context":      metadata.GetExtendedSchema(getContextSchema(true, false, true)),
```
[the remark this answers is not in the document]
Comment (ksamoray) [reply nsxt/resource_nsxt_vpc_subnet.go:34]: We could IMO, VPC object contexts are always required. IDK if the generator distinguishes VPC objects which require `context` attribute and those which require `parent_path`. @annakhm is this doable?

File: nsxt/resource_nsxt_vpc_subnet.go (hunk 1 above)
Comment (annakhm) [reply nsxt/resource_nsxt_vpc_subnet.go:216]: This fix is already present in generator - based on whether `parent_path` is specified or not, the correct importer should be chosen

**Review by annakhm — APPROVED** [review]

**Pull request merged**